src/NuGet.Core/NuGet.Packaging.Extraction/NuGet.Packaging.Extraction.csproj

```diff
@@ -307,6 +363,26 @@
     </EmbeddedResource>
   </ItemGroup>
 
+  <Target Name="ILMergeNPE" Condition="'$(BuildingInsideVisualStudio)' != 'true' and '$(SkipILMergeOfNPE)' != 'true'" >
+    <ItemGroup>
+      <BuildArtifacts Include="$(OutputPath)\*.dll" Exclude="@(MergeExclude)" />
+    </ItemGroup>
+    <PropertyGroup>
+      <PathToBuiltNPE>$(OutputPath)NuGet.Packaging.Extraction.dll</PathToBuiltNPE>
+      <IlmergeCommand>$(ILMergeExePath) /lib:$(OutputPath) /t:library /out:$(ILMergeDirectory)NuGet.Packaging.Extraction.dll @(MergeAllowDup -> '/allowdup:%(Identity)', ' ')</IlmergeCommand>
+      <IlmergeCommand Condition="Exists($(MS_PFX_PATH))">$(IlmergeCommand) /delaysign /keyfile:$(MS_PFX_PATH)</IlmergeCommand>
+      <IlmergeCommand>$(IlmergeCommand) $(PathToBuiltNPE) @(MergeInclude->'%(fullpath)', ' ')</IlmergeCommand>
+    </PropertyGroup>
+    <MakeDir Directories="$(ILMergeDirectory)" />
+    <Exec Command="$(IlmergeCommand)" ContinueOnError="false" StandardOutputImportance="High" StandardErrorImportance="High" />
+  </Target>
+
   <Import Project="$(BuildCommonDirectory)common.targets" />
   <Import Project="Sdk.targets" Sdk="Microsoft.NET.Sdk" />
+
+  <PropertyGroup>
```

Comment:
You mentioned that disabling the framework references didn't work...one thing to make sure is that

https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.CommandLine/NuGet.CommandLine.csproj#L156 is added *after* the common.targets, cause the common targets import the pack SDK.

Reply:
isn't it more proper to keep them, in this case?

Reply:
yeah, you're right, this will be a library :D 
I'm sure you mentioned this somewhere but I missed it.